common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -117,7 +116,7 @@ public ShuffleIndexInformation load(File file) throws IOException {
   private AppShufflePartitionInfo getOrCreateAppShufflePartitionInfo(
       AppShuffleId appShuffleId,
       int reduceId) {
-    File dataFile = getMergedShuffleDataFile(appShuffleId, reduceId);
+    File dataFile = getMergedShuffleDataFile(appShuffleId.appId, appShuffleId.shuffleId, reduceId);
```

Comment:
When the server receives a pushBlock message then there are 2 cases:
1. It belongs to previous attempt: in this case it needs to ignore that message. If we don't then it will corrupt the file here.
2. It belongs to latest attempt: in this case we should continue merging

Reply:
Added AttemptID check here. If the attemptID is different with the one recorded in AppAttemptPathInfo, it will return null and will take this block as a late block, so it will be ignored.

Reply:
Will resolve the multi-attempts in SPARK-35546